src/Sarif.UnitTests/SarifTransformerTests.cs

```diff
@@ -168,5 +170,93 @@ public void SarifTransformerTests_ToCurrent_OneRunWithLogicalLocations()
             logicalLocations["collections"].Kind.Should().Be("namespace");
             logicalLocations["collections"].ParentKey.Should().BeNull();
         }
+
+        [Fact]
+        public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
+        {
+            string content = "The quick brown fox jumps over the lazy dog";
+            byte[] bytes = Encoding.ASCII.GetBytes(content);
+            string v1LogText =
+              @"{
+                  ""version"": ""1.0.0"",
+                  ""runs"": [
+                    {
+                      ""files"": {
+                        ""file:///home/list.txt"": {
+                          ""mimeType"": ""text/plain"",
+                          ""length"": 43,
+                          ""contents"": ""VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZw=="",
+                          ""hashes"": [
+                            {
+                              ""algorithm"": ""sha256"",
+                              ""value"": ""d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592""
+                            }
+                          ]
+                        },
+                        ""file:///home/buildAgent/bin/app.zip"": {
+                          ""mimeType"": ""application/zip""
+                        },
+                        ""file:///home/buildAgent/bin/app.zip#/docs/intro.docx"": {
+                          ""uri"": ""file:///docs/intro.docx"",
+                          ""mimeType"": ""application/vnd.openxmlformats-officedocument.wordprocessingml.document"",
+                          ""parentKey"": ""file:///home/buildAgent/bin/app.zip"",
+                          ""offset"": 17522,
+                          ""length"": 4050
+                        }
+                      },
+                      ""tool"": {
+                        ""name"": ""CodeScanner"",
+                        ""semanticVersion"": ""2.1.0""
+                      },
+                      ""results"": [
+                      ]
+                    }
+                  ]
+                }";
+
+            SarifLogVersionOne v1Log = GetSarifLogVersionOne(v1LogText);
+            var transformer = new SarifVersionOneToCurrentVisitor();
+            transformer.VisitSarifLogVersionOne(v1Log);
+
+            SarifLog v2Log = transformer.SarifLog;
+
+            v2Log.Runs.Should().NotBeNull();
+            v2Log.Runs.Count.Should().Be(1);
+
+            Dictionary<string, FileData> files = v2Log.Runs[0].Files as Dictionary<string, FileData>;
```

Comment:
Cast is not needed. `Files` is already `IDictionary<string, FileData>`. Your test code shouldn't need a concrete `Dictionary`.